### PR TITLE
Adds an RGB upgrade project

### PR DESCRIPTION
**Pull request by maxion12345**
# obama out

# Document the changes in your pull request

AI's can now research and make cool RGB lighting
# Wiki Documentation
we have a wiki page for this?

# Changelog

Edit the changelog below to reflect the changes made by this PR, even if the changes are minor - required for every PR.
If you add a name after the ':cl', that name will be used in the changelog. Leave it empty to use your GitHub name.

:cl:  
rscadd: AI's can now research RGB lightings for their datacores. 
/:cl:

### CODE REVIEW

**Comment by bruhlookatthisdood** [comment]
[Common] MAL-F-TI..N alarms, "IM HOSTING A RAD RAVE IN MY SAT!!"

File: code/modules/mob/living/silicon/ai/decentralized/ai_data_core.dm [hunk 1]
```diff
@@ -143,7 +145,15 @@ GLOBAL_VAR_INIT(primary_data_core, null)
 		var/mutable_appearance/on_overlay = mutable_appearance(icon, "[initial(icon_state)]_on")
 		add_overlay(on_overlay)
 
+/obj/machinery/ai/data_core/proc/partytime()
+	var/current_color = random_color()
+	set_light(7, 3, current_color)
+	TimerID = addtimer(CALLBACK(src, .proc/partytime), 5, TIMER_STOPPABLE)
```
Comment (Bibby0110) [inline code/modules/mob/living/silicon/ai/decentralized/ai_data_core.dm]:
```suggestion
	TimerID = addtimer(CALLBACK(src, .proc/partytime), 0.5 SECONDS, TIMER_STOPPABLE)
```

File: code/modules/mob/living/silicon/ai/decentralized/projects/rgb.dm [hunk 2]
```diff
@@ -0,0 +1,26 @@
+/datum/ai_project/rgb
+	name = "RGB Lighting"
+	description = "By varying the current levels in the lighting subsystems of your servers, you can make pretty colors."
+	research_cost = 250
+	ram_required = 0
+	research_requirements_text = "None"
+	category = AI_PROJECT_MISC
+	var/current_color
```
Comment (Bibby0110) [inline code/modules/mob/living/silicon/ai/decentralized/projects/rgb.dm]: unused

File: code/modules/mob/living/silicon/ai/decentralized/projects/rgb.dm [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+		datacores.partytime()
+
+
+/datum/ai_project/rgb/stop()
+	for(var/obj/machinery/ai/data_core/datacores in GLOB.data_cores)
+		datacores.stoptheparty()
+	..()
+
+		
+
```
Comment (Bibby0110) [inline code/modules/mob/living/silicon/ai/decentralized/projects/rgb.dm]: empty lines

**Review by Bibby0110 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/silicon/ai/decentralized/ai_data_core.dm [hunk 4]
```diff
@@ -143,7 +145,15 @@ GLOBAL_VAR_INIT(primary_data_core, null)
 		var/mutable_appearance/on_overlay = mutable_appearance(icon, "[initial(icon_state)]_on")
 		add_overlay(on_overlay)
 
+/obj/machinery/ai/data_core/proc/partytime()
+	var/current_color = random_color()
```
Comment (Bibby0110) [inline code/modules/mob/living/silicon/ai/decentralized/ai_data_core.dm:151]: check to see if we're already partying

**Review by Bibby0110 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/silicon/ai/decentralized/ai_data_core.dm [hunk 5]
```diff
@@ -143,7 +145,17 @@ GLOBAL_VAR_INIT(primary_data_core, null)
 		var/mutable_appearance/on_overlay = mutable_appearance(icon, "[initial(icon_state)]_on")
 		add_overlay(on_overlay)
 
-
+/obj/machinery/ai/data_core/proc/partytime()
+	if(TimerID)
+		return 0
```
Comment (Bibby0110) [inline code/modules/mob/living/silicon/ai/decentralized/ai_data_core.dm]:
```suggestion
		return FALSE
```

**Review by Bibby0110 — APPROVED** [review]

**Pull request merged**